### DISCUSSION TITLE
StreamDeck, Mobiflight and FSUIPC

### DISCUSSION THREAD

**Discussion by imodude** [nguyenquyhy/Flight-Tracker-StreamDeck#251] [General]
I understand that this is a little off topic since it spans more subjects than just this awesome StreamDeck module, but...

I've only just got my streamdeck, so am trying to read all about the great work people have been doing to interface it to MSFS2020.

Heck it's confusing!

I know I must have mobiflight installed and setup right because my profiles are correctly  activating the GNS buttons (e.g. MOBIFLIGHT_AS530_RightLargeKnob_Right).

But I'm really not getting how to add extra mappings....
If the latest mobiflight event.txt file has, for example, 
A32NX_ADIRS_KNOB_1_NAV#1 (>L:A32NX_ADIRS_KNOB_1)

How do I make use of this in Flight-Tracker-StreamDeck? 
I guess it's not just a simple case of adding it to TOGGLE_EVENTS.cs.

Do I need to do some mapping in the MobiFlight gui app? 
I've tried following some youtube tutorials but they're mostly concerned with connecting to Arduino boards, not StreamDeck, so I dont really follow. And I cant follow very far because my "modules" drop down is empty.

Any tips appreciated - apologies for being off topic :-)

Regards, Ian

**Comment by Slotherby** [comment]
@imodude Did you solve this issue?
I am new to this and really struggling with how to use this for the A32NX Dev.
I have the latest MobiFlight WASM module installed and it works great for some switches.
eg Generic Toggle Event MOBIFLIGHT_A32NX_OVHD_FUELSYSTEM_LTKPUMP1_TOGGLE

However I cannot get the ADIRS knobs to work, both of these are listed in the MobiFlight events.txt file but neither works
MOBIFLIGHT_A32NX_ADIRS_KNOB_1_NAV
MOBIFLIGHT_OVHD_ADIRS_KNOB1_NAV

Any help would be much appreciated

Rgds
Paul

**Reply by nguyenquyhy** [reply]
Are you using version 0.39 or are you using the fork from imodude? If you are using 0.39, please also try `Mobiflight.A32NX_ADIRS_KNOB_1_NAV`.

**Reply by Slotherby** [reply]
Thanks for the reply.

Somewhere during my troubleshooting the problem was fixed.  I installed the latest version of MobiFlight (9.1) and the latest version of their WASM module (0.3.33).  I also made sure that I was running version 0.39.
I now can manipulate the ADIRS knobs using MOBIFLIGHT_OVHD_ADIRS_KNOB1_NAV
I was also able to get the Battery switches working but this time I had to use  Mobiflight.OVHD_ELEC_BATTERY_1_TOGGLE

Should there be any difference between MOBIFLIGHT_ and Mobiflight.  ?
Am I correct in my understanding that if an event is listed in the WASM module events.txt file then I should be able to access it in FlightTracker by just using MOBIFLIGHT_ or Mobiflight.  ?

Thanks again for your awesome program.  It is making my experience of MSFS2020 so much better.

Paul

**Reply by nguyenquyhy** [reply]
> Should there be any difference between MOBIFLIGHT_ and Mobiflight. ?

You can always use `MobiFlight.`. I think that is what it is supposed to be. `MOBIFLIGHT_` was just some unintended consequences of previous implementation of Flight Tracker plugin. In the new implementation I translate `MOBIFLIGHT_` to `MobiFlight.` to maintain backward compatibility.
